invocation.md

```diff
@@ -0,0 +1,1170 @@
+  value T
+}
+```
+
+Above schema compiled to standard syntax
+
+```ipldsch
+type Box struct {
+  value Box_T
+}
+type Box_T any
+```
+
+## 1.5 Signatures
+
+All payloads described in this spec MUST be signed with a [Varsig].
+
+# 2 High-Level Concepts
+
+## 2.1 Roles
+
+Task adds two new roles to UCAN: invoker and executor. The existing UCAN delegator and delegate principals MUST persist to the invocation.
+
+| UCAN Field | Delegation                             | Task                            |
+| ---------- | -------------------------------------- | ------------------------------- |
+| `iss`      | Delegator: transfer authority (active) | Invoker: request task (active)  |
+| `aud`      | Delegate: gain authority (passive)     | Executor: perform task (active) |
+
+### 2.1.1 Invoker
+
+The invoker signals to the executor that a task associated with a UCAN SHOULD be performed.
+
+The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `iss` field of the contained UCAN.
+
+### 2.1.2 Executor
+
+The executor is directed to perform some task described in the UCAN by the invoker.
+
+The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
+
+## 2.2 Components
+
+```mermaid
+flowchart LR
+ U(UCAN) -- Auth --> I
+ I(Invocation) -- Output --> O(Result)
+ O -- Consume --> P(Promise)
+ O -- Attest --> R(Receipt)
+ R -- Short-circuit --> M(Memoization)
+```
+
+### 2.2.1 Invocation
+
+An [Invocation] is like a cryptographically signed function application, a request to perform some action on a resource with specific input. Invocation MAY have optional metadata that is not used to describe the meaning of the computation or effect to be run. Executor MUST produce same result regardless of the metadata.
+
+### 2.2.2 Receipt
+
+A [Receipt] describes the output of an invocation. It is referenced either by the invocation itself or an IPLD Link of the invocation.
+
+### 2.2.3 Promise
+
+A [promise] is a reference to expected result of the invocation receipt.
+
+## 2.3 IPLD Schema
+
+```ipldsch
+type Invocation<In> struct {
+  v      SemVer
+
+  iss    Principal
+  aud    Principal
+
+  with   URI
+  do     Ability
+  input  In (implicit {})
+
+  meta   {String : Any} (implicit {})
+
+  prf    [&UCAN]
+  nnc    optional String
+  nbf    optional Int
+
+  s      Varsig
+}
+
+type Receipt<In, Out> struct {
+  job     &Invocation<In>
+
+  # output of the invocation
+  out     Out
+
+  # Related receipts
+  origin optional &Receipt<In, Any>
+
+  # All the other metadata
+  meta    { String: Any }
+
+  # Principal that issued this receipt
+  iss     Principal
+
+  # Signature from the `iss`.
+  s       Varsig
+
+  # Proof that `iss` was authorized to by invocation `aud` to issue
+  # receipts. Can be omitted if `job.aud === this.iss`
+  prf     [&UCAN] implicit ([])
+}
+
+# Promise is a way to reference data from the output of the invocation
+type Promise struct {
+  Await    "<-"
```

Comment:
A minor point because it's just the characters. I initially really liked the pun on `<-`, but the overwhelming feedback internally at Fission is that it's difficult (or at least surprising and unfamialiar) to read. Using words instead of symbols may be a good idea

Reply:
Yeah I'm not going to die on this hill, if `<-` controversial I'm fine with whatever. Although I do find argument bit moot, it's something in binary format that is nothing anyone will ever manually type or read, in fact (other than pun) name was chosen is it would be very unlikely to collide with any other key someone might actually have on their structs.